src/main/java/net/dv8tion/jda/api/utils/messages/MessageCreateBuilder.java

```diff
@@ -0,0 +1,288 @@
+        return new MessageCreateBuilder().applyEditData(data);
+    }
+
+    /**
+     * Factory method to start a builder from an existing instance of {@link Message}.
+     * <br>Equivalent to {@code new MessageCreateBuilder().applyMessage(data)}.
+     * <br>The {@link AllowedMentions} are not updated to reflect the provided message, and might mention users that the message did not.
+     *
+     * <p>This cannot copy the file attachments of the message, they must be manually downloaded and provided to {@link #setFiles(FileUpload...)}.
+     *
+     * @param  message
+     *         The message data to apply
+     *
+     * @throws IllegalArgumentException
+     *         If the message is null or a system message
+     *
+     * @return A new MessageCreateBuilder instance with the applied data
+     *
+     * @see    #applyMessage(Message)
+     */
+    @Nonnull
+    public static MessageCreateBuilder fromMessage(@Nonnull Message message)
+    {
+        return new MessageCreateBuilder().applyMessage(message);
+    }
+    
+    @Nonnull
+    @Override
+    public MessageCreateBuilder addContent(@Nonnull String content)
+    {
+        Checks.notNull(content, "Content");
+        Checks.check(
+            Helpers.codePointLength(this.content) + Helpers.codePointLength(content) <= Message.MAX_CONTENT_LENGTH,
+            "Cannot have content longer than %d characters", Message.MAX_CONTENT_LENGTH
+        );
+        this.content.append(content);
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public MessageCreateBuilder addEmbeds(@Nonnull Collection<? extends MessageEmbed> embeds)
+    {
+        Checks.noneNull(embeds, "Embeds");
+        Checks.check(
+            this.embeds.size() + embeds.size() <= Message.MAX_EMBED_COUNT,
+            "Cannot add more than %d embeds", Message.MAX_EMBED_COUNT
+        );
+        this.embeds.addAll(embeds);
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public MessageCreateBuilder addComponents(@Nonnull Collection<? extends LayoutComponent> components)
+    {
+        Checks.noneNull(components, "ComponentLayouts");
+        for (LayoutComponent layout : components)
+            Checks.check(layout.isMessageCompatible(), "Provided component layout is invalid for messages!");
+        Checks.check(
+                this.components.size() + components.size() <= Message.MAX_COMPONENT_COUNT,
+            "Cannot add more than %d component layouts", Message.MAX_COMPONENT_COUNT
+        );
+        this.components.addAll(components);
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public MessageCreateBuilder setFiles(@Nullable Collection<? extends FileUpload> files)
+    {
+        if (files != null)
+            Checks.noneNull(files, "Files");
+        this.files.clear();
+        if (files != null)
+            this.files.addAll(files);
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public List<? extends FileUpload> getAttachments()
```

Comment:
shouldn't this be `getFiles`?

Reply:
This is declared on the abstract interface, so it also has to support attachment updates.